web/html/src/components/table/SearchField.tsx

```diff
@@ -1,17 +1,30 @@
 import * as React from "react";
 
+import ReactSelect from "react-select";
+
 type SearchFieldProps = {
-  /** Search criteria value */
+  /** The value that the user will enter to perform the filter on the criteriaField */
   criteria?: string;
 
+  /** The field on which the user like to perform the filter */
+  field?: string | any;
+
+  /** This is the set of options that will be displayed */
+  options?: any;
```

Comment:
Even if we don't know the types, it would be nice to annotate that this is an array as in `options?: any[];`

Reply:
Here these are what is eaten by `ReactSelect` or `Select` if we manage to switch to it

Reply:
Just added a new type for it